plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java

```diff
@@ -159,7 +217,25 @@ String selectSql(TableId table, String formattedColumns)
 
     private String fullTableName(TableId tableId)
     {
-        tableId = tableIds.getOrDefault(tableId, tableId);
+        String remoteSchemaName = toRemoteDatasetName(tableId.getProject(), tableId.getDataset());
+        String remoteTableName = toRemoteTableName(tableId.getProject(), remoteSchemaName, tableId.getTable());
+        tableId = TableId.of(tableId.getProject(), remoteSchemaName, remoteTableName);
         return format("%s.%s.%s", tableId.getProject(), tableId.getDataset(), tableId.getTable());
     }
+
+    List<BigQueryColumnHandle> getColumns(BigQueryTableHandle tableHandle)
+    {
+        TableInfo tableInfo = getTable(tableHandle.getRemoteTableName().toTableId());
+        Schema schema = tableInfo.getDefinition().getSchema();
+        if (schema == null) {
+            throw new TableNotFoundException(
+                    tableHandle.getSchemaTableName(),
+                    format("Table '%s' not found or has no supported columns", tableHandle.getSchemaTableName()));
```

Comment:
If my understanding is correct, we will get NPE at L228 in case of table not found. Is it wrong?